test/SlowTests/Issues/RavenDB-16614.cs

```diff
@@ -0,0 +1,329 @@
+            using var store = GetDocumentStore(new Options { Server = leader, ReplicationFactor = 3 });
+
+            using (var session = store.OpenAsyncSession(new SessionOptions { TransactionMode = TransactionMode.ClusterWide }))
+            {
+                await session.StoreAsync(new User { Name = "arava" }, "users/arava");
+                await session.StoreAsync(new User { Name = "phoebe" }, "users/phoebe");
+                await session.SaveChangesAsync();
+            }
+
+            using (var session = store.OpenAsyncSession(new SessionOptions { TransactionMode = TransactionMode.ClusterWide }))
+            {
+                var user = await session.LoadAsync<User>("users/arava");
+                user.Name = "Arava Eini";
+                var user2 = await session.LoadAsync<User>("users/phoebe");
+                user2.Name = "Phoebe Eini";
+
+                using (var conflictedSession = store.OpenAsyncSession(new SessionOptions {TransactionMode = TransactionMode.ClusterWide}))
+                {
+                    var conflictedArava = await conflictedSession.LoadAsync<User>("users/arava");
+                    conflictedArava.Name = "Arava!";
+                    await conflictedSession.SaveChangesAsync();
+                }
+
+                await Assert.ThrowsAsync<ConcurrencyException>(() => session.SaveChangesAsync());
+            }
+        }
+
+
+        [Fact]
+        public async Task ModificationInAnotherTransactionWillFailWithDelete()
+        {
+            var leader = await CreateRaftClusterAndGetLeader(1);
+            using var store = GetDocumentStore(new Options { Server = leader, ReplicationFactor = 3 });
+
+            using (var session = store.OpenAsyncSession(new SessionOptions { TransactionMode = TransactionMode.ClusterWide }))
+            {
+                await session.StoreAsync(new User { Name = "arava" }, "users/arava");
+                await session.StoreAsync(new User { Name = "phoebe" }, "users/phoebe");
+                await session.SaveChangesAsync();
+            }
+
+            using (var session = store.OpenAsyncSession(new SessionOptions { TransactionMode = TransactionMode.ClusterWide }))
+            {
+                var user = await session.LoadAsync<User>("users/arava");
+                session.Delete(user);
+                var user2 = await session.LoadAsync<User>("users/phoebe");
+                user2.Name = "Phoebe Eini";
+
+                using (var conflictedSession = store.OpenAsyncSession(new SessionOptions { TransactionMode = TransactionMode.ClusterWide }))
+                {
+                    var conflictedArava = await conflictedSession.LoadAsync<User>("users/arava");
+                    conflictedArava.Name = "Arava!";
+                    await conflictedSession.SaveChangesAsync();
+                }
+
+                await Assert.ThrowsAsync<ConcurrencyException>(() => session.SaveChangesAsync());
+            }
+        }
+    }
```

Comment:
Can we have some tests for import / export, backup / restore, replication?

Reply:
We have `WillDeleteOrphanedAtomicGuards_AfterRestoreFromBackup` for backups / restore.
Replication is tested: `CanHandleConflictsWithClusterTransactionIndex`